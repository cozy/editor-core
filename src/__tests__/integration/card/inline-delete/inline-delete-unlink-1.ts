import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import Page from '@atlaskit/webdriver-runner/wd-wrapper';
import {
  getDocFromElement,
  editable,
  copyToClipboard,
  gotoEditor,
} from '../../_helpers';
import { waitForInlineCardSelection } from '@atlaskit/media-integration-test-helpers';

BrowserTestCase(
  `card: unlinking an Inline Link should replace it with text corresponding to the title of the previously linked page`,
  {
    skip: ['safari', 'edge'],
  },
  async (client: ConstructorParameters<typeof Page>[0], testName: string) => {
    const page = new Page(client);

    // Copy stuff to clipboard
    await copyToClipboard(page, 'https://www.atlassian.com');

    await gotoEditor(page);

    // Paste the link
    await page.paste();

    await waitForInlineCardSelection(page);
    await page.click('button[aria-label="Unlink"]');

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);
