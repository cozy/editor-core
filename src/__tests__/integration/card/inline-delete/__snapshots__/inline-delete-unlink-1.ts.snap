// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card: unlinking an Inline Link should replace it with text corresponding to the title of the previously linked page 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "text": "Welcome to Atlassian! ",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;
